### DISCUSSION TITLE
Change of behaviour for default styling

### DISCUSSION THREAD

**Discussion by alderg** [jgraph/drawio#3312] [General]
We will be changing the default behaviour of styling from the 21.0.0 release. The current behaviour is as follows:

- There are two stored styles, one for shapes and one for connectors.
- When you make changes to certain styles on a shape or connector, the default, stored, style for that type is updated. The font characteristics used in a shape become the default for the next shape, for example.
- There are also options to set the default style based on the current selection. Those are available in the format panel and right-click menu, depending on the interface used.

![Screenshot 2023-01-24 at 11 02 45](https://user-images.githubusercontent.com/1761087/214263411-ad3226cf-b1c7-46fc-9440-d9705978e82c.png)

From 21.0.0 onwards you will need to explicitly set default style(s) using the interface. The current behaviour is non-standard and we prefer to avoid "hidden" behaviours like this one.

If you prefer the current behaviour, there is a [configuration option](https://www.diagrams.net/doc/faq/configure-diagram-editor) to replace it:

```
{
  "updateDefaultStyle": true
}
```
You can set that behaviour by clicking this link (**don't click the link if you don't want it**) : https://app.diagrams.net/#_CONFIG_UzV3UjUyKi1ISSxJdUlNSyzNKQkuqcxJBQqqGjuWFJWmqpq7AAA=

That will add the entry to your JSON config.

**Comment by davidjgraph** [comment]
This change comes into effect in today's 21.0.1 release.
